### ISSUE TITLE
Trying to run the google-sheets-sync example fails, "No public directory specified" when a public directory is in fact specified

### ISSUE THREAD

**Issue by MasonP57** [firebase/functions-samples#271]
### How to reproduce these conditions
https://github.com/firebase/functions-samples/tree/master/google-sheet-sync
clone this firebase/functions-samples repo
navigate to google-sheet-sync
$ firebase init
Follow the procedure
$ firebase deploy

Fails
At first I got HTTP 403
After exploring some settings and some more googling, I restarted from scratch. Followed the instructions above exactly. When it asked to overwrite the package.json, I allowed it. Then I went into the difference via VS Code and copied over the google dependencies from the original example.

Now when I run $ firebase deploy it tells me
"Error: No public directory specified, can't deploy hosting"

firebase.json:
"functions": {
    "source": "functions"
  }

"functions" is the name of the directory with index.js, all of this is straight from this github repo.

I think this google sheets example is out of date, and therefore no longer working correctly. Which is a shame because it's the only resource I can find about using Firebase Functions to connect to a Google API and I need this capability.

**Sample name or URL where you found the bug**

**Failing Function code used (including require/import commands at the top)**

**Steps to set up and reproduce**

**Sample data pasted or attached as JSON (not an image)**

**Security rules used**

### Debug output

**Errors in the [console logs](https://console.firebase.google.com/project/_/functions/logs?search=&severity=DEBUG)**

**Screenshots**

**Comment by prescottprue** [comment]
Looks like your hosting isn't deploying since you don't have whatever folder the Firebase config for hosting is pointing to. You should try to only deploy functions (i.e. `firebase deploy --only functions`) and see if that helps.

**Comment by MasonP57** [comment]
Thanks for the reply, I have managed to fix the error. I still had the Firebase tutorial for uploading to Functions that worked so I compared the files and fiddled with them. In the end I deleted this repo, re-cloned it. Installed the google-auth-library and googleapis via npm, made the firebase.json file an empty object like in the project that works and now it's working.

I've been able to deploy the google-sheets-sync example from this repo so I guess this issue can be closed.
